### PR TITLE
Upgrade tokio to version 1.0.0

### PR DESCRIPTION
**Pull request by Tim-Zhang**
Fixes: #4

Signed-off-by: Tim Zhang <tim@hyper.sh>

### CODE REVIEW

File: src/stream.rs [hunk 1]
```diff
@@ -43,117 +43,165 @@
...
+        unsafe {
+            b = &mut *(buf.unfilled_mut() as *mut [mem::MaybeUninit<u8>] as *mut [u8]);
+        };
+
+        loop {
+            let mut guard = ready!(self.inner.poll_read_ready(cx))?;
+
+            match guard.try_io(|inner| inner.get_ref().read(b)) {
+                Ok(result) => match result {
+                    Ok(n) => {
```
Comment (mxpv) [inline src/stream.rs:169]: double match is a bit hard to read.
How do you feel about `if let Some` instead?
```rust
            if let Some(result) = guard.try_io(|inner| inner.get_ref().read(b)) {
                return match result {
                    Ok(n) => {
                        unsafe {
                            buf.assume_init(n);
                        }
                        buf.advance(n);
                        Ok(()).into()
                    }
                    Err(e) => Err(e).into(),
                };
            }
```

**Review by mxpv — COMMENTED** [review]
This is great! Minor suggestion (repeats for multiple places)

File: src/stream.rs (hunk 1 above)
Comment (Tim-Zhang) [reply src/stream.rs]: @mxpv Thank you very much for your review and good advice.

I have changed the code to this 
```
match guard.try_io(|inner| inner.get_ref().read(b)) {
    Ok(Ok(n)) => {
        unsafe {
            buf.assume_init(n);
        }
        buf.advance(n);
        return Ok(()).into();
    }
    Ok(Err(e)) => return Err(e).into(),
    Err(_would_block) => {
        continue;
    }
}
```

What about this style?

**Comment by Tim-Zhang** [comment]
@dpeckett Please have a look for this pr

File: src/stream.rs (hunk 1 above)
Comment (mxpv) [reply src/stream.rs]: nice, didn't know that pattern matcher in rust that smart :)

**Review by shanecurran — APPROVED** [review]

**Comment by Tim-Zhang** [comment]
@shanecurran Can you please merge it and cargo publish it? thank you very much.

**Comment by shanecurran** [comment]
@Tim-Zhang Unfortunately I don't have write permissions on the repo, but your changes look good to me. Hopefully @mxpv can approve and merge shortly!

**Comment by mxpv** [comment]
@dpeckett PTAL

**Comment by shanecurran** [comment]
Just wanted to bump this @mxpv @dpeckett

Thank you!

**Comment by Tim-Zhang** [comment]
@shanecurran @dpeckett has transferred this repo to the new org rust-vsock. Welcome to join the org, I have invited you.

I plan to merge this and publish 0.3.0 next week

**Review by shanecurran — APPROVED** [review]

**Comment by Tim-Zhang** [comment]
It's ready to merge now, I will merge it tomorrow and release the 0.3.0

**Pull request merged**